rfcs/severity.md

```diff
@@ -0,0 +1,130 @@
+so we provide more actionable feedback and help developers be more succesfull.
+
+## Proposal
+
+**TL;DR;**
+
+Have hints report with the same severity accross all the different mediums (cli,
+vs code, browser extension, and online scanner) by:
+
+1. Changing the available severity levels for hints.
+1. Making hints responsible of setting the severity for each report.
+1. Letting users change only if a hint is enable or disable (plus some extra
+   configuration in the applicable hints).
+
+### Hint severity level
+
+The idea is to get closer to the severity options available in
+[VS Code](https://code.visualstudio.com/api/references/vscode-api#DiagnosticSeverity):
+
+| Severity | Description |
+| -------- | ----------- |
+| off      | Disabled, no reporting will come from it |
+| error    | Something that the user should get fixed right away |
+| warning  | Something that the developer should probably look into |
+| hint*    | Something minor to look into if you have time |
+| information* | FYI for the user |
+
+The new values are `hint` and `information`.
+
+**Note:** Not sure about the `information` value. Will revisit once we have the severity
+for each hint message
+
+### Hint severity
+
+As mentioned previously, there are a few hints that report multiple things were the
+relevance is not the same. An example would be `http-compression` where we report about:
+
+* Not using compression
+* Using compression on binaries
+* Not using brotli
+* Not respecting `Accept-Encoding: identity`
+
+Probably the first 2 should be `error`, `brotli` a `warning` or a `hint` and the one about
+`identity` a `hint` or `information`.
+
+Having all the reports with the same severity does not help the user focus on what's
+important (have compression enabled for only text based resources
+([see GitHub comment](https://github.com/webhintio/hint/issues/2919#issuecomment-530190038))
+
+**Other examples here**
+
+### User configuration
+
+With the proposed changes, a user will only be able to decide which hints are enabled or not.
+The user configuration will then look like this:
+
+```json
+{
+    "connector": {
+        "name": "puppeteer",
+        "options": {
+            "waitUntil": "networkidle2"
+        }
+    },
+    "hintsTimeout": 120000,
+    "hints": {
+        "amp-validator": "off",
+        "apple-touch-icons": "on",
+        "button-type": "off",
+        "content-type": "on",
+        ...
+        "no-vulnerable-javascript-libraries": ["on", {
+            "severity": "high"
+        }]
+    }
+}
+```
+
+As a shortcut, the property `hints` could accept an array of hint names that
+will be turned on:
+
+```json
+{
+    "connector": {
+        "name": "puppeteer",
+        "options": {
+            "waitUntil": "networkidle2"
+        }
+    },
+    "hintsTimeout": 120000,
+    "hints": [
+        "apple-touch-icons",
+        "content-type",
+        ...
+        "no-vulnerable-javascript-libraries"
+        ]
+    }
+}
+```
```

Comment:
One thing I forgot to add is that a user should be now capable of setting a threshold for the severity and how that impacts the exit code.
E.g.: A user might want to be notified on `warning` and higher and ignore everything else. The process exit code should be different than 0 even if there are only warnings.

What about a `threshold` property with `"error|warning|hint|information"` as possible values?